### PR TITLE
Refactor testing

### PR DESCRIPTION
**Pull request by chrisjsewell**
- move from Travis to Github Actions for CI
- use RTD for documentation build previews (see https://docs.readthedocs.io/en/stable/guides/autobuild-docs-for-pull-requests.html)
- moved from `pdflatex` to `xelatex` for pdf builds: it is more modern and handles UTF8 better (pdflatex was failing)
- removed `sphinx.ext.imgmath`, to use `sphinx.ext.mathjax` instead (for math rendering)
- added to pre-commit
  - check-yaml
  - end-of-file-fixer
  - trailing-whitespace
  - [rstcheck](https://github.com/myint/rstcheck)
  - [markdownlint](https://github.com/markdownlint/markdownlint) 
- fixed any failing tests

### CODE REVIEW

**Comment by chrisjsewell** [comment]
Just waiting on https://github.com/aiidateam/aiida-core/pull/4203#issuecomment-650309665 for final fix

**Comment by chrisjsewell** [comment]
@ltalirz don't know if you realised, but https://github.com/aiidateam/aiida-tutorials/pull/130 actually just completely stopped transifex from working lol: https://travis-ci.org/github/aiidateam/aiida-tutorials/jobs/675701667 (it just silently fails)

**Comment by ltalirz** [comment]
> @ltalirz don't know if you realised, but #130 actually just completely stopped transifex from working

Oops , thanks for noticing ;-)

File: docs/conf.py [hunk 1]
```diff
@@ -358,7 +356,8 @@ def setup(app):
 
 # Intersphinx configuration
 intersphinx_mapping = {
-    'aiida': ('http://aiida-core.readthedocs.io/en/latest/', None),
+    'aiida':
+    ('http://aiida-core.readthedocs.io/en/fix-4200-read-the-docs/', None),
```
Comment (chrisjsewell) [inline docs/conf.py]: note this needs to be changed back to `latest` before merging, once https://github.com/aiidateam/aiida-core/pull/4203 is merged

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      run: |
+        make -C docs ${{ matrix.format }}
+      env:
+        SPHINXOPTS: -nW --keep-going
+    - name: "build docs: latexpdf"
+      if: ${{ matrix.format == 'latexpdf' }}
+      run: |
+        make -C docs ${{ matrix.format }}
+      # TODO there are reference issues for latexpdf in e.g. docs/pages/2019_molsim_school_Amsterdam/screening/export.md
+      # I believe due to relative document references which don't work in latexpdf, since all documents are merged into one doctree
```
Comment (chrisjsewell) [inline .github/workflows/ci.yml:78]: added this an issue #230 to be fixed separately

**Comment by ltalirz** [comment]
Thanks, changes look fine to me - I guess you are waiting until https://github.com/aiidateam/aiida-core/pull/4203 ; feel free to go ahead with merging this afterwards.

**Pull request merged**